compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java

```diff
@@ -198,8 +198,19 @@ public static String cleanupPathSeparators(String name) {
         return name.replace(WINDOWS_PATH_SEPERATOR, JAVA_PACKAGE_SEPERATOR);
     }
 
-    public static String rewriteVirtualCallTypeName(String value) {
-        return Utils.encodeFunctionIdentifier(cleanupObjectTypeName(value));
+    public static String rewriteVirtualCallTypeName(String value, BType objectType) {
+        objectType = getImpliedType(objectType);
+        // For attached functions from another module the call name will be in the format `objectTypeName.funcName`.
+        // We need to remove the type name.
+        if (!objectType.tsymbol.name.value.isEmpty() && value.startsWith(objectType.tsymbol.name.value)) {
+            value = value.replace(objectType.tsymbol.name.value + ".", "").trim();
+        }
+        // For attached functions from another module where the type is an anonType, the call name will be in
+        // the format `(objectTypeName).funcName`. We need to remove the type name.
+        if (value.startsWith("(") && value.contains(").")) {
+            value = value.substring(value.indexOf(").") + 2);
```

Comment:
Any idea why this was not covered?

Reply:
This is not covered as the symbol name is already filtered at the BIRGen. Therefore we don't need to analyse it here.
 https://github.com/ballerina-platform/ballerina-lang/blob/10fc9656e84eb8734128fe58bf2e6bf1e71c3b93/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java#L892
Therefore, I removed this check and sent the same for 2201.9.x as well.
https://github.com/ballerina-platform/ballerina-lang/pull/43170